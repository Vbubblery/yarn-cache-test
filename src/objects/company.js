import { validate } from "jsonschema";
import CompanySchema from "@/schemas/company";

export class Company {
  constructor(model) {
    this._id = model.id;
    this._name = model.name;
    this._autoFillSupply = model.auto_fill_supply;
    this._users = model.users ? model.users.map(user => user.id) : [];
    this._sites = model.sites ? model.sites : [];
    this._allowSftpExport = model.allow_sftp_export;
  }
  static create(model) {
    return validate(model, Company.schema()) ? new Company(model) : null;
  }
  static schema() {
    return CompanySchema;
  }

  // Getters
  get id() {
    return this._id;
  }
  get name() {
    return this._name;
  }
  get autoFillSupply() {
    return this._autoFillSupply;
  }
  get users() {
    return this._users;
  }
  get sites() {
    return this._sites;
  }
  get allowSftpExport() {
    return this._allowSftpExport;
  }

  // Setters
  set id(id) {
    this._id = id;
  }
  set name(name) {
    this._name = name;
  }
  set autoFillSupply(autoFillSupply) {
    this._autoFillSupply = autoFillSupply;
  }
  set users(users) {
    this._users = users;
  }
  set sites(sites) {
    this._sites = sites;
  }
  set allowSftpExport(value) {
    throw new Error(
      "This operation is not allowed. Please contact Flowlity to enable sftp export for your company"
    );
  }
}
