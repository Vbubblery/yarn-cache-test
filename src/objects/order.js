import { validate } from "jsonschema";
import OrderSchema, { orderSupplySchema } from "@/schemas/order";

export class OrderSupply {
  constructor(model = {}) {
    this._model = model;
    this._id = model.id;
    this._productId = model.product_site;
    this._externalId = model.external_id;
    this._isDefaultUnit = model.is_default_unit;
    this._defaultUnitValue = model.default_unit_value;
    this._quantity = model.quantity;
    this._secondaryUnitValue = model.second_unit_value;
    this._secondaryUnitQuantity = model.second_unit_quantity;
    this._orderId = model.order;
    this._name = model.name;
    this._unit = model.unit;
    this._secondaryUnit = model.second_unit;
    this._leadTime = model.lead_time;
    this._price = model.product_price;
    this._totalPrice = model.total_price;
    this._moq = model.moq;
    this._description = model.description;
    this._lotSize = model.lot_size;
    this._stockAtEndOfPeriod = model.stock_at_end_of_period;
    this._minStockAtEndOfPeriod = model.minimum_stock_at_end_of_period;
    this._totalDemand = model.total_demand;
  }
  static create(model) {
    return validate(model, OrderSupply.schema())
      ? new OrderSupply(model)
      : null;
  }

  static schema() {
    return orderSupplySchema;
  }

  get externalId() {
    return this._externalId;
  }

  get defaultUnitValue() {
    return this._defaultUnitValue;
  }

  get secondaryUnitValue() {
    return this._secondaryUnitValue;
  }

  get secondaryUnitQuantity() {
    return this._secondaryUnitQuantity;
  }

  get secondaryUnit() {
    return this._secondaryUnit;
  }

  get moq() {
    return this._moq;
  }

  get description() {
    return this._description;
  }

  get lotSize() {
    return this._lotSize;
  }

  get id() {
    return this._id;
  }

  get productId() {
    return this._productId;
  }

  get quantity() {
    return this._quantity;
  }

  get orderId() {
    return this._orderId;
  }

  get name() {
    return this._name;
  }

  get unit() {
    return this._unit;
  }

  get leadTime() {
    return this._leadTime;
  }

  get price() {
    return this._price;
  }

  get totalPrice() {
    return this._totalPrice;
  }

  get isDefaultUnit() {
    return this._isDefaultUnit;
  }

  get minStockAtEndOfPeriod() {
    return this._minStockAtEndOfPeriod;
  }
  get stockAtEndOfPeriod() {
    return this._stockAtEndOfPeriod;
  }
  get totalDemand() {
    return this._totalDemand;
  }
  set id(id) {
    this._id = id;
  }

  set productId(productId) {
    this._productId = productId;
  }

  set orderId(orderId) {
    this._orderId = orderId;
  }

  set quantity(quantity) {
    this._quantity = quantity;
  }

  set name(name) {
    this._name = name;
  }

  set unit(unit) {
    this._unit = unit;
  }

  set leadTime(leadTime) {
    this._leadTime = leadTime;
  }
  set totalPrice(totalPrice) {
    this._totalPrice = totalPrice;
  }

  add(otherSupply) {
    return new OrderSupply({
      ...this._model,
      quantity: this.quantity + otherSupply.quantity
    });
  }
}

export class Order {
  constructor(model = {}) {
    this._customerSiteId = model.customer_site_id;
    this._customerSiteName = model.customer_site_name;
    this._date = model.date;
    this._externalId = model.external_id;
    this._fullTruckQuantity = model.full_truck_quantity;
    this._fullTruckUnit = model.full_truck_unit;
    this._id = model.id;
    this._minOrderQuantity = model.minimum_order_quantity;
    this._minOrderUnit = model.minimum_order_unit;
    this._orderFrequency = model.order_frequency;
    this._previousOrderDate = model.previous_order_date;
    this._sendBefore = model.send_before;
    this._shipBefore = model.ship_before;
    this._statusId = model.status_id;
    this._supplierId = model.supplier_id;
    this._supplies = (model.supplies || []).map(supply =>
      OrderSupply.create(supply)
    );
    this._supplierName = model.name;
    this._tagGrouper = model.tag_grouper;
    this._leadTime = model.lead_time;
    this._lastSftpExportDate = model.last_sftp_export_date;
  }
  static create(model) {
    return validate(model, Order.schema()) ? new Order(model) : null;
  }
  static schema() {
    return OrderSchema;
  }

  // Getters
  get customerSiteId() {
    return this._customerSiteId;
  }
  get customerSiteName() {
    return this._customerSiteName;
  }
  get orderFrequency() {
    return this._orderFrequency;
  }
  get previousOrderDate() {
    return this._previousOrderDate;
  }
  get sendBefore() {
    return this._sendBefore;
  }
  get shipBefore() {
    return this._shipBefore;
  }
  get id() {
    return this._id;
  }
  get statusId() {
    return this._statusId;
  }
  get date() {
    return this._date;
  }
  get supplierId() {
    return this._supplierId;
  }
  get supplies() {
    return this._supplies;
  }
  get fullTruckQuantity() {
    return this._fullTruckQuantity;
  }
  get fullTruckUnit() {
    return this._fullTruckUnit;
  }
  get minOrderQuantity() {
    return this._minOrderQuantity;
  }
  get minOrderUnit() {
    return this._minOrderUnit;
  }
  get supplierName() {
    return this._supplierName;
  }
  get externalId() {
    return this._externalId;
  }
  get tagGrouper() {
    return this._tagGrouper;
  }
  get leadTime() {
    return this._leadTime;
  }
  get lastSftpExportDate() {
    return this._lastSftpExportDate;
  }

  // Setters
  set id(id) {
    this._id = id;
  }

  set statusId(statusId) {
    this._statusId = statusId;
  }
  set date(date) {
    this._date = date;
  }
  set supplierId(supplierId) {
    throw new Error(
      "Cannot change a supplier for an existing order. Create a new order instead"
    );
  }
  set supplies(supplies) {
    this._supplies = supplies;
  }
  set fullTruckQuantity(fullTruckQuantity) {
    this._fullTruckQuantity = fullTruckQuantity;
  }
  set fullTruckUnit(fullTruckUnit) {
    this._fullTruckUnit = fullTruckUnit;
  }
  set minOrderQuantity(minOrderQuantity) {
    this._minOrderQuantity = minOrderQuantity;
  }
  set minOrderUnit(minOrderUnit) {
    this._minOrderUnit = minOrderUnit;
  }
  set supplierName(supplierName) {
    this._supplierName = supplierName;
  }
  set externalId(externalId) {
    this._externalId = externalId;
  }
  set tagGrouper(tagGrouper) {
    this._tagGrouper = tagGrouper;
  }
  set lastSftpExportDate(lastSftpExportDate) {
    this._lastSftpExportDate = lastSftpExportDate;
  }
}
