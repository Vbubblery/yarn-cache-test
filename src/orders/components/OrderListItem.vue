<template>
  <div
    class="oli bg-white px-2 flex items-center border-b-2 border-dashed cursor-pointer w-full min-h-[60px] group"
    data-testid="order_list_item_container"
    @click="showDetails"
  >
    <div class="w-8 max-w-8 h-full flex items-center">
      <BaseTooltip
        v-if="canUserAccessToOrderWarnings"
        light
        data-testid="order_alert_tooltip"
        placement="top"
      >
        <BaseSVG name="error-warning-fill" class="text-orange-500" />
        <template v-if="!isOrderValid" #content>
          <div v-if="invalidProductsCount > 0" class="oli__warning">
            <p class="font-base leading-6 font-medium">
              {{ $t("orders.productIssues") }}
            </p>
            <p>
              <span class="text-orange-600">
                {{ invalidProductsCount }}
              </span>
              {{ $tc("orders.issueMOQ", invalidProductsCount) }}
            </p>
          </div>
          <div v-if="invalidProductsCount === 0" class="oli__warning">
            <p class="oli__warning font-base leading-6 font-medium">
              {{ $t("orders.orderIssues") }}
            </p>
            <p
              v-if="!isOrderRespectingMOQConstraint && order.minOrderUnit"
              class="oli__warning text-orange-600"
            >
              {{ $t("orders.constraintMOQ") }} (
              <span class="text-orange-600">
                {{ order.minOrderQuantity }}
                {{ order.minOrderUnit.toLowerCase() }} </span
              >)
            </p>
            <p
              v-if="
                !isOrderRespectingOrderFrequencyConstraint &&
                order.orderFrequency
              "
              class="oli__warning"
            >
              {{ $t("orders.constraintFrequency") }} (
              <span class="text-orange-600">
                {{ order.orderFrequency }}
                {{ $t("days") }}</span
              >)
            </p>
            <p
              v-if="
                !isOrderRespectingFullTruckQuantityConstraint &&
                order.fullTruckUnit
              "
              class="oli__warning"
            >
              {{ $t("orders.constraintFullTruck") }} (<span
                class="text-orange-600"
                >{{ order.fullTruckQuantity }}
                {{ order.fullTruckUnit.toLowerCase() }}</span
              >)
            </p>
          </div>
        </template>
      </BaseTooltip>
    </div>
    <div class="text-left items-center justify-start inline-grid px-2 group">
      <input
        type="checkbox"
        class="h-4 w-4"
        :checked="isSelected"
        @click.stop="selectOrderById(order.id)"
      />
    </div>
    <div class="grid grid-cols-9 gap-2 content-center text-sm w-full py-2 mx-3">
      <div class="inline-grid items-center col-span-1 px-4">
        <p>
          {{
            [3, 4].includes(orderTypeValue) && order.externalId
              ? order.externalId
              : order.id
          }}
        </p>
      </div>
      <div
        v-if="[1, 2].includes(orderTypeValue)"
        class="col-span-1 inline-grid items-center px-1"
        @click.self="showDetails"
      >
        <p>{{ dateToDisplay }}</p>
      </div>
      <div class="col-span-1 inline-grid items-center px-1">
        <p v-if="dateLabelToDisplay !== 'date'" class="text-gray-600">
          {{ deliveryDate }}
        </p>
      </div>
      <div class="inline-grid items-center px-1" :class="supplierHeadingClass">
        {{ order.supplierName }}
      </div>
      <div class="col-span-1 inline-grid items-center px-2">
        <p class="text-gray-700 text-right">
          {{ supplies.length }}
          <span class="text-gray-600 text-2xs">{{
            $tc("orders.items", supplies.length)
          }}</span>
        </p>
      </div>
      <div class="col-span-1 inline-grid items-right px-2">
        <p class="text-gray-700 text-right py-2">
          {{ formattedSuppliesTotal }}
          <span class="text-gray-600 text-2xs">â‚¬</span>
        </p>
      </div>
      <div class="col-span-1 inline-grid items-center px-3 text-right">
        <div
          :class="{
            'group-hover:hidden': isActionAvailable && !showSFTPExportDateColumn
          }"
          class="block text-gray-700 py-1"
        >
          <span v-if="isTotalQuantityPrimaryValid && quantity.primaryUnit">
            {{ quantity.primaryUnitQuantity }}
            <span class="text-gray-600 text-2xs">
              {{ quantity.primaryUnit.toLowerCase() }}
            </span>
          </span>
          <span class="mx-2">/</span>
          <span v-if="isTotalQuantitySecondaryValid && quantity.secondaryUnit">
            {{ quantity.secondaryUnitQuantity }}
            <span class="text-gray-600 text-2xs">
              {{ quantity.secondaryUnit.toLowerCase() }}
            </span>
          </span>
        </div>

        <OrderListItemActions
          v-if="!showSFTPExportDateColumn"
          :available-actions="availableActions"
          @order:accept="acceptCurrentOrder"
          @order:discard="discardCurrentOrder"
        />
      </div>
      <div
        v-if="showSFTPExportDateColumn"
        class="col-span-1 inline-grid items-center px-3 text-right"
      >
        <div
          :class="{ 'group-hover:hidden': isActionAvailable }"
          class="block text-gray-700 py-1"
        >
          {{ lastSftpExportDate }}
        </div>

        <OrderListItemActions
          :available-actions="availableActions"
          @order:accept="acceptCurrentOrder"
          @order:discard="discardCurrentOrder"
        />
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import helper from "@/helper/helper";
import { Order, OrderSupply } from "@/objects/order";
import BaseTooltip from "@/components/contents/common/BaseTooltip.vue";
import BaseSVG from "@/components/base/components/BaseSVG.vue";
import { Site } from "@/objects/site";
import {
  computed,
  defineComponent,
  SetupContext,
  toRefs
} from "@vue/composition-api";
import { ORDER_ACTION_TYPE, ORDER_TYPE_VALUE } from "@/orders/constants";
import { useOrderActions, useOrderDetails } from "@/orders/composables";
import OrderListItemActions from "@/orders/components/OrderListItemActions.vue";
import { Company } from "@/objects/company";

export default defineComponent({
  components: {
    OrderListItemActions,
    BaseTooltip,
    BaseSVG
  },
  props: {
    isSelected: {
      type: Boolean,
      required: false,
      default: false
    },
    shouldDisplayActions: {
      type: Boolean,
      required: false
    },
    orderTypeValue: {
      type: Number,
      required: true
    },
    availableActions: {
      type: Array,
      required: true
    },
    order: {
      type: Object as () => Order,
      required: true
    },
    dateLabelToDisplay: {
      type: String,
      required: false,
      default: "date"
    }
  },
  emits: ["select:order", "update:viewItem", "reload:page"],
  setup(props, context: SetupContext) {
    const { availableActions, order, orderTypeValue } = toRefs(props);

    const currentSite = computed<Site>(
      () => context.root.$store.getters["common/getCurrentSite"]
    );

    const company = computed<Company>(
      () => context.root.$store.getters["common/getCompany"]
    );

    const {
      quantity,
      formattedSuppliesTotal,
      isOrderRespectingOrderFrequencyConstraint,
      isOrderRespectingFullTruckQuantityConstraint,
      isOrderRespectingMOQConstraint
    } = useOrderDetails(order);

    const supplies = computed<OrderSupply[]>(() => {
      return props.order?.supplies || [];
    });

    const deliveryDate = computed<string>(() => {
      return helper.switchDateFormat(
        props.order?.date,
        "YYYY-MM-DD",
        "DD/MM/YYYY"
      );
    });

    const { acceptCurrentOrder, discardCurrentOrder } = useOrderActions(
      context,
      order,
      currentSite
    );

    const dateToDisplay = computed<string>(() => {
      let date;
      switch (props.dateLabelToDisplay) {
        case "send_before":
          date = order.value?.sendBefore;
          break;
        case "ship_before":
          date = order.value?.shipBefore;
          break;
        default:
          date = order.value?.date;
          break;
      }
      return helper.switchDateFormat(date, "YYYY-MM-DD", "DD/MM/YYYY");
    });

    const invalidProductsCount = computed<number>(() => {
      return (order.value?.supplies || []).filter(
        (supply: OrderSupply) =>
          (supply.moq && supply.quantity < supply.moq) ||
          (supply.lotSize && supply.quantity % supply.lotSize !== 0)
      ).length;
    });

    const isOrderValid = computed<boolean>(() => {
      return (
        isOrderRespectingMOQConstraint.value &&
        isOrderRespectingFullTruckQuantityConstraint.value &&
        invalidProductsCount.value === 0 &&
        isOrderRespectingOrderFrequencyConstraint.value
      );
    });

    const isTotalQuantityPrimaryValid = computed<boolean>(() => {
      const primaryUnits = (props.order?.supplies || []).map(
        (supply: OrderSupply) => supply.unit
      );
      const primaryUnitsSet = new Set(primaryUnits);
      return (
        primaryUnitsSet.size === 1 &&
        !primaryUnitsSet.has("") &&
        !primaryUnitsSet.has(null)
      );
    });

    const isTotalQuantitySecondaryValid = computed<boolean>(() => {
      const secondaryUnits = (props.order?.supplies || []).map(
        (supply: OrderSupply) => supply.secondaryUnit
      );
      const secondaryUnitsSet = new Set(secondaryUnits);
      return (
        secondaryUnitsSet.size === 1 &&
        !secondaryUnitsSet.has("") &&
        !secondaryUnitsSet.has(null)
      );
    });

    const showDetails = () => {
      context.emit("update:viewItem", order.value.id);
    };

    const selectOrderById = (id: string) => {
      context.emit("select:order", { id });
    };

    const canUserAccessToOrderWarnings = computed(
      () => !currentSite.value.isSupplierView && !isOrderValid.value
    );

    const isActionAvailable = computed(
      () =>
        availableActions.value.includes(ORDER_ACTION_TYPE.CONFIRM) ||
        availableActions.value.includes(ORDER_ACTION_TYPE.UNCONFIRMED)
    );

    const showSFTPExportDateColumn = computed<boolean>(() => {
      return (
        !currentSite.value.isSupplierView &&
        orderTypeValue.value === ORDER_TYPE_VALUE.VALIDATED &&
        company.value.allowSftpExport
      );
    });

    const lastSftpExportDate = computed<string>(() => {
      if (order.value.lastSftpExportDate) {
        return helper.toCalendarDate(order.value.lastSftpExportDate);
      } else {
        return "â€“";
      }
    });

    const supplierHeadingClass = computed<string>(() => {
      const isTemporaryStatus = [
        ORDER_TYPE_VALUE.PLANNED,
        ORDER_TYPE_VALUE.VALIDATED
      ].includes(orderTypeValue.value);

      if (isTemporaryStatus) {
        return showSFTPExportDateColumn.value ? "col-span-2" : "col-span-3";
      } else {
        return showSFTPExportDateColumn.value ? "col-span-3" : "col-span-4";
      }
    });

    return {
      acceptCurrentOrder,
      discardCurrentOrder,
      currentSite,
      showDetails,
      selectOrderById,
      supplies,
      deliveryDate,
      dateToDisplay,
      formattedSuppliesTotal,
      isOrderRespectingOrderFrequencyConstraint,
      quantity,
      isOrderRespectingFullTruckQuantityConstraint,
      isOrderRespectingMOQConstraint,
      invalidProductsCount,
      isOrderValid,
      isTotalQuantityPrimaryValid,
      isTotalQuantitySecondaryValid,
      canUserAccessToOrderWarnings,
      isActionAvailable,
      ORDER_TYPE_VALUE,
      showSFTPExportDateColumn,
      lastSftpExportDate,
      supplierHeadingClass
    };
  }
});
</script>
<style lang="scss" scoped>
input {
  width: 20px;
  height: 20px;
}

.oli {
  &__warning {
    margin-bottom: 4px;

    &:last-child {
      margin-bottom: 0;
    }
  }
}
</style>
