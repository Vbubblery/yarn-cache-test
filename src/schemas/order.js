export default {
  type: "object",
  properties: {
    id: {
      type: "integer",
      minimum: 1
    },
    status_id: {
      type: "integer",
      minimum: 1,
      maximum: 4
    },
    supplier_id: {
      type: "integer",
      minimum: 1
    },
    date: {
      type: "string"
    },
    supplies: {
      type: "object"
    }
  },
  required: ["id", "status_id", "supplier_id", "date", "supplies"]
};

export const orderSupplySchema = {
  type: "object",
  properties: {
    id: {
      type: "integer",
      minimum: 1
    },
    product_site: {
      type: "integer",
      minimum: 1
    },
    quantity: {
      type: "integer"
    },
    order: {
      type: "integer",
      minimum: 1
    },
    name: {
      type: "string"
    }
  },
  required: ["id", "product_site", "quantity", "name"]
};
