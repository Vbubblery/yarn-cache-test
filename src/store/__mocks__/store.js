import Vue from "vue";
import Vuex from "vuex";

import common from "./common/common";
import data from "./data/data";
import planning from "./planning/planning";
import orders from "./orders/orders";
import demand from "./demand/demand";
import alerts from "./alerts/alerts";
import dashboard from "./dashboard/dashboard";

Vue.use(Vuex);

export const state = {};

export const getters = {};

export const modules = {
  data,
  common,
  planning,
  orders,
  demand,
  alerts,
  dashboard
};

export const mutations = {
  reset: jest.fn()
};

export const actions = {};

export function createMocks(
  custom = { state: {}, getters: {}, mutations: {}, actions: {}, modules: {} }
) {
  const mockState = Object.assign({}, state, custom.state);
  const mockGetters = Object.assign({}, getters, custom.getters);
  const mockMutations = Object.assign({}, mutations, custom.mutations);
  const mockActions = Object.assign({}, actions, custom.actions);
  const mockModules = Object.assign({}, modules, custom.modules);

  return {
    state: mockState,
    getters: mockGetters,
    mutations: mockMutations,
    actions: mockActions,
    modules: mockModules,
    store: new Vuex.Store({
      state: mockState,
      getters: mockGetters,
      mutations: mockMutations,
      actions: mockActions,
      modules: mockModules
    })
  };
}

export const store = createMocks().store;
