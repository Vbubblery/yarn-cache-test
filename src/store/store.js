import Vue from "vue";
import Vuex from "vuex";
// Modules
import common from "./common/common";
import data from "./data/data";
import planning from "./planning/planning";
import orders from "./orders/orders";
import demand from "./demand/demand";
import alerts from "./alerts/alerts";
import dashboard from "./dashboard/dashboard";

Vue.use(Vuex);
let store;
try {
  store = new Vuex.Store({
    modules: {
      data,
      common,
      planning,
      orders,
      demand,
      alerts,
      dashboard
    },
    mutations: {
      reset(state) {
        Object.keys(state).forEach(key => {
          Object.assign(state[key], initialState[key]);
          if (!Object.prototype.hasOwnProperty.call(initialState, key)) {
            delete state[key];
          }
        });
      }
    }
  });
  const initialState = JSON.parse(JSON.stringify(store.state));
  // eslint-disable-next-line no-empty
} catch (error) {}

export { store };
