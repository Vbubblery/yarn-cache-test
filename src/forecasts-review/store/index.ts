import { computed, SetupContext, ComputedRef } from "@vue/composition-api";
import { PLANNING_DEMAND_VIEW } from "@/forecasts-review/typings";
import { Site } from "@/objects/site";

export const useBaseResources = (context: SetupContext) => {
  const store = context.parent?.$store || null;

  const getMe = computed<{ locale: string }>(
    () => store?.getters["common/getMe"]
  );
  const planningDemandView: ComputedRef<PLANNING_DEMAND_VIEW> = computed(
    () => store?.getters["common/getDefaultView"]
  );
  const currentSite = computed<Site>(
    () => store?.getters["common/getCurrentSite"]
  );
  const tablesData = computed(() => store?.getters["demand/getTablesData"]);
  const demandGetChartData = computed(
    () => store?.getters["demand/getChartData"]
  );
  const demandGetIsLoadingChart = computed(
    () => store?.getters["demand/getIsLoadingChart"]
  );
  const demandGetIsLoadingTables = computed(
    () => store?.getters["demand/getIsLoadingTables"]
  );
  const planningGetChartData = computed(
    () => store?.getters["planning/getChartData"]
  );
  const planningGetIsLoadingChart = computed(
    () => store?.getters["planning/getIsLoadingChart"]
  );
  const planningGetIsLoadingTables = computed(
    () => store?.getters["planning/getIsLoadingTables"]
  );
  const selectedTagIds = computed(
    () => store?.getters["common/getSelectedTags"]
  );
  const selectedCustomerId = computed(() =>
    currentSite.value.isSupplierView
      ? store?.getters["demand/getSelectedCustomerSiteId"]
      : null
  );
  const updateDefaultView = (view: PLANNING_DEMAND_VIEW) => {
    store?.commit("common/updateDefaultView", view);
  };
  const updateSelectedProducts = (items: object[]) => {
    store?.dispatch("common/updateSelectedProducts", items);
  };
  const updateSelectedTags = (items: object[]) => {
    store?.dispatch("common/updateSelectedTags", items);
  };
  const updateDataType = (itemType: string) => {
    store?.commit("common/updateDataType", itemType);
  };
  const resetDemandState = () => {
    store?.commit("demand/resetState");
  };
  const updateDemandPage = (pageNum: number) => {
    store?.commit("demand/updatePage", pageNum);
  };
  const updateDemandTableOnly = (onlyTable: boolean) => {
    store?.commit("demand/updateOnlyTable", onlyTable);
  };
  const fetchData = (
    selectedView: undefined | PLANNING_DEMAND_VIEW = undefined
  ) => {
    const currentView = selectedView ?? planningDemandView.value;
    store?.dispatch(
      currentView === PLANNING_DEMAND_VIEW.DEMAND
        ? "demand/fetchDemandData"
        : "planning/updatePlanningData"
    );
  };
  const updateSelectedCustomerSiteId = (customerSiteId: number | null) => {
    store?.commit("demand/updateSelectedCustomerSiteId", customerSiteId);
  };
  return {
    getMe,
    planningDemandView,
    updateDefaultView,
    currentSite,
    tablesData,
    demandGetChartData,
    demandGetIsLoadingChart,
    demandGetIsLoadingTables,
    planningGetChartData,
    planningGetIsLoadingChart,
    planningGetIsLoadingTables,
    selectedCustomerId,
    selectedTagIds,
    updateSelectedProducts,
    updateSelectedTags,
    updateDataType,
    resetDemandState,
    updateDemandPage,
    updateDemandTableOnly,
    fetchData,
    updateSelectedCustomerSiteId
  };
};
