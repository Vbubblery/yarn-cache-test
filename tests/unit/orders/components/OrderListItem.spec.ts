import { renderComponent } from "tests/test-utils";
import OrderListItem from "@/orders/components/OrderListItem.vue";
// eslint-disable-next-line jest/no-mocks-import
import { store } from "@/store/__mocks__/store";
import { Order } from "@/objects/order";
describe("Orders Filter Header", () => {
  it("Renders the component - even with empty order", () => {
    const { html } = renderComponent(OrderListItem, {
      store,
      propsData: {
        orderTypeValue: 1,
        availableActions: [],
        order: new Order()
      }
    });
    expect(html()).toMatchSnapshot();
  });
  it("Displays alert icon if order does not respect all constraints - MOQ", async () => {
    const { findByTestId } = renderComponent(OrderListItem, {
      store,
      propsData: {
        orderTypeValue: 1,
        availableActions: [],
        order: new Order({
          minimum_order_unit: "eur",
          minimum_order_quantity: 10
        })
      }
    });
    await expect(findByTestId(/order_alert_tooltip/)).resolves.toBeTruthy();
  });
  it("Displays alert icon if order does not respect all constraints - Full Truck", async () => {
    const { findByTestId } = renderComponent(OrderListItem, {
      store,
      propsData: {
        orderTypeValue: 1,
        availableActions: [],
        order: new Order({
          full_truck_quantity: 10,
          full_truck_unit: "",
          supplies: [{ quantity: 1 }]
        })
      }
    });
    await expect(findByTestId(/order_alert_tooltip/)).resolves.toBeTruthy();
  });
  it("Displays alert icon if order does not respect all constraints - Products count", async () => {
    const { findByTestId } = renderComponent(OrderListItem, {
      store,
      propsData: {
        orderTypeValue: 1,
        availableActions: [],
        order: new Order({
          supplies: [{ quantity: 1, moq: 2 }]
        })
      }
    });
    await expect(findByTestId(/order_alert_tooltip/)).resolves.toBeTruthy();
  });
  it("Displays alert icon if order does not respect all constraints - Frequency", async () => {
    const { findByTestId } = renderComponent(OrderListItem, {
      store,
      propsData: {
        orderTypeValue: 1,
        availableActions: [],
        order: new Order({
          previous_order_date: "fakeDate",
          order_frequency: 10
        })
      }
    });
    await expect(findByTestId(/order_alert_tooltip/)).resolves.toBeTruthy();
  });
});
