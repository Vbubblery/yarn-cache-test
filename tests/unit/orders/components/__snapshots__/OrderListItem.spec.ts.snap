// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders Filter Header Renders the component - even with empty order 1`] = `
<div data-testid="order_list_item_container" class="oli bg-white px-2 flex items-center border-b-2 border-dashed cursor-pointer w-full min-h-[60px] group">
  <div class="w-8 max-w-8 h-full flex items-center">
    <!---->
  </div>
  <div class="text-left items-center justify-start inline-grid px-2 group"><input type="checkbox" class="h-4 w-4"></div>
  <div class="grid grid-cols-9 gap-2 content-center text-sm w-full py-2 mx-3">
    <div class="inline-grid items-center col-span-1 px-4">
      <p>

      </p>
    </div>
    <div class="col-span-1 inline-grid items-center px-1">
      <p>Invalid Date</p>
    </div>
    <div class="col-span-1 inline-grid items-center px-1">
      <!---->
    </div>
    <div class="inline-grid items-center px-1 col-span-3">

    </div>
    <div class="col-span-1 inline-grid items-center px-2">
      <p class="text-gray-700 text-right">
        0
        <span class="text-gray-600 text-2xs">orders.items</span>
      </p>
    </div>
    <div class="col-span-1 inline-grid items-right px-2">
      <p class="text-gray-700 text-right py-2">
        0,00
        <span class="text-gray-600 text-2xs">â‚¬</span>
      </p>
    </div>
    <div class="col-span-1 inline-grid items-center px-3 text-right">
      <div class="block text-gray-700 py-1">
        <!----> <span class="mx-2">/</span>
        <!---->
      </div>
      <div class="flex justify-end">
        <!---->
      </div>
    </div>
    <!---->
  </div>
</div>
`;
